frontend/src/utils/hooks/usePagination.tsx

```diff
@@ -0,0 +1,113 @@
+import { DocumentNode, useQuery } from "@apollo/client";
+import React, { useEffect, useState } from "react";
+import Paginator from "../../components/utils/Paginator";
+
+interface PaginatedResult<T> {
+  totalCount: number;
+  edges: PaginatedEdge<T>[];
+  pageInfo: PageInfo;
+}
+
+interface PaginatedEdge<T> {
+  cursor: string;
+  node: T;
+  pageInfo: PageInfo;
+}
+
+interface PageInfo {
+  endCursor: string;
+  hasNextPage: boolean;
+}
+
+/*
+ * A hook that adapts cursor-style pagination to support a table-based use-case.
+ * @param query - The paginated GraphQL query
```

Comment:
nit: is it worth mentioning it's relationship with relay?

Reply:
sure, i'll modify the comment!